cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -0,0 +1,214 @@
+        }
+    }
+
+    private KafkaConfiguration emptyKafkaConf() {
+        Map<String, Object> conf = new HashMap<>();
+        return new KafkaConfiguration(conf.entrySet());
+    }
+
+
+    private void fillPlaceholderValue(Map<String, String> orderedProperties, String placeholder, String value) {
+        orderedProperties.entrySet().forEach(entry -> {
+            entry.setValue(entry.getValue().replaceAll("\\$\\{" + Pattern.quote(placeholder) + "\\}", value));
+        });
+    }
+
+    public KafkaConfiguration getDiff() {
+        return this.diff;
+    }
+
+    public boolean isDesiredPropertyDefaultValue(String key) {
+        Optional<ConfigEntry> entry = currentEntries.stream().filter(configEntry -> configEntry.name().equals(key)).findFirst();
+        if (entry.isPresent()) {
+            return entry.get().isDefault();
+        }
+        return false;
+    }
+
+    public boolean cannotBeUpdatedDynamically() {
+        if (diff == null) {
+            return false;
+        } else return diff.anyReadOnly(kafkaVersion)
+                || !diff.unknownConfigs(kafkaVersion).isEmpty()
+                || diff.containsListenersChange();
+    }
+
+    /**
+     * @return Map object which is used for dynamic configuration of kafka broker
+     */
+    public Map<ConfigResource, Collection<AlterConfigOp>> getUpdatedConfig() {
+        return updated;
+    }
+
+    private boolean isIgnorableProperty(String key) {
+        return IGNORABLE_PROPERTIES.matcher(key).matches();
+    }
+
+    /**
+     * Computes diff between two maps. Entries in IGNORABLE_PROPERTIES are skipped
+     * @return KafkaConfiguration containing all entries which were changed from current in desired configuration
+     */
+    public KafkaConfiguration computeDiff() {
+        Map<String, String> currentMap = new HashMap<>();
+        Map<String, String> difference = new HashMap<>();
+
+        Collection<AlterConfigOp> updatedCE = new ArrayList<>();
+
+        currentEntries.stream().forEach(e -> {
+            currentMap.put(e.name(), e.value());
+        });
+
+        Map<String, String> desiredMap = ModelUtils.configMap2Map(desired, "server.config");
+
+        fillPlaceholderValue(desiredMap, "STRIMZI_BROKER_ID", Integer.toString(brokerId));
+
+        JsonNode source = patchMapper().valueToTree(currentMap);
+        JsonNode target = patchMapper().valueToTree(desiredMap);
+        JsonNode jsonDiff = JsonDiff.asJson(source, target);
+
+        for (JsonNode d : jsonDiff) {
+            String pathValueWithoutSlash = d.get("path").asText().substring(1);
+            String pathValue = d.get("path").asText();
+
+            Optional<ConfigEntry> entry = currentEntries.stream().filter(e -> e.name().equals(pathValueWithoutSlash)).findFirst();
+
+            if (entry.isPresent()) {
+                if (d.get("op").asText().equals("remove")) {
+                    if (!entry.get().source().name().equals("DEFAULT_CONFIG")) {
```

Comment:
Why not compare with the `ConfigSource.DEFAULT_CONFIG` enum member, rather than converting to text?

Also, missing Trace logging in this block?